### PR TITLE
Added guess_bonds (fixes Issue #245)

### LINKED ISSUE #245: Guess bonds on Universe creation
Currently PDBParsers have a guess_bonds keyword which lets them guess bonds on construction.  It would be good to move this out of PDB and available for all formats.

So I could do..
```python
u = mda.Universe('something.gro', guess_bonds=True)
```
And have bonds guessed.

What would be the best way to do this?

Adding a keyword to Universe init to guess bonds once top and trj and loaded?
```python
u = mda.Universe(top, trj, guess_bonds=True)
```
Add a method to Universe which guesses bonds?
This has an advantage as you could pass custom vdwradii
```python
u.guess_bonds(radii = ?)
```
Or extend this to include bonds, angles, torsions & impropers
```python
u = mda.Universe(top, trj, guess_topology=True)
u.guess_topology()
```
Or all of the above?

The guessing functions exist are in topology.core, so this should be something easy to include, just wanted some feedback on what looks best.

### PR DESCRIPTION
**Pull request by richardjgowers**
Ok, tests pass, ready to merge pending review.

I think there's a merge conflict because of CHANGELOG

(Will fix #245)

### CODE REVIEW

File: package/MDAnalysis/core/AtomGroup.py [hunk 1]
```diff
@@ -3123,17 +3235,13 @@ def __init__(self, *args, **kwargs):
              individual list member. [``None``]
              Can also pass a subclass of :class:`MDAnalysis.coordinates.base.Reader`
              to define a custom reader to be used on the trajectory file.
-          *bonds*
```
Comment (orbeckst) [inline package/MDAnalysis/core/AtomGroup.py]: Mention in CHANGELOG that the bonds keyword went away and was replaced by guess_bonds.

For right now, catch `bonds` keyword, issue a deprecation warning ("will go away in 0.11.0"), and try to do the right thing. We then remove that code for 0.11.0.

File: package/MDAnalysis/core/AtomGroup.py [hunk 2]
```diff
@@ -3228,6 +3339,16 @@ def __init__(self, *args, **kwargs):
         # Load coordinates
         self.load_new(coordinatefile, **kwargs)
 
+        # Deprecated bonds mode handling here, remove eventually.
+        if 'bonds' in kwargs:
+            raise DeprecationWarning(
```
Comment (orbeckst) [inline package/MDAnalysis/core/AtomGroup.py]: Shouldn't this be

```python
warnings.warn("The 'bonds' keyword has been deprecated and will be removed in release 0.11.0. Use 'guess_bonds' instead.")
```

File: package/MDAnalysis/core/AtomGroup.py [hunk 3]
```diff
@@ -3296,42 +3443,19 @@ def _init_bonds(self):
            Now only accepts list of tuples as input, previously accepted either
```
Comment (orbeckst) [inline package/MDAnalysis/core/AtomGroup.py]: The "previously accepted" doc seems irrelevant; if at all put in a versionchanged markup.

File: package/MDAnalysis/core/AtomGroup.py [hunk 4]
```diff
@@ -1116,15 +1230,15 @@ def bonds(self):
         only bonds which are entirely contained by the AtomGroup use
         u.bonds.atomgroup_intersection(ag, strict=True)
 
-        .. versionadded 0.9.0
+        .. versionadded:: 0.9.0
+        .. versionchanged:: 0.9.2
```
Comment (orbeckst) [inline package/MDAnalysis/core/AtomGroup.py]: Replace all the versionchanged:: 0.9.2 with 0.10.0 (I think – this might be a sufficiently big change to go to 0.10.0)

File: testsuite/MDAnalysisTests/test_atomgroup.py [hunk 5]
```diff
@@ -1798,3 +1798,86 @@ def test_wrap_fragments(self):
...
+        assert_equal(len(u.angles), 2)
+        assert_equal(len(u.torsions), 0)
+        assert_equal(len(u.atoms[0].bonds), 2)
+        assert_equal(len(u.atoms[1].bonds), 1)
+        assert_equal(len(u.atoms[2].bonds), 1)
+        assert_equal(len(u.atoms[3].bonds), 2)
+        assert_equal(len(u.atoms[4].bonds), 1)
+        assert_equal(len(u.atoms[5].bonds), 1)
+
+    def test_universe_guess_bonds(self):
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/test_atomgroup.py]: Also add a check for `guess_bonds = False`.

File: package/CHANGELOG [hunk 6]
```diff
@@ -20,6 +20,12 @@ The rules for this file:
...
   * Added wrap method. Wrap allows the centers of groups of atoms to be
     moved within the primary unit cell without breaking up the group. (Issue 190)
+  * Added guess_bonds keyword argument to Universe creation.  This will attempt to
+    guess all topology information on Universe creation. (Issue #245)
+  * Added guess_bonds method to AtomGroup. (Issue #245)
+  * All TopologyObjects (Bond, Angle etc) now have is_guessed attribute
+  * TopologyGroup now has alternate constructor method, .from_indices()
+  * Added TopologyObject.indices property
 
   Changes
```
Comment (orbeckst) [inline package/CHANGELOG]: deprecated 'bond' keyword and replaced with 'guess_bonds'

**Comment by orbeckst** [comment]
This looks good, @richardjgowers . Comments inline, all pretty harmless.
I think this feature will trigger a bump for 0.10.0 because it introduces a new and potentially big feature – @seb-buch what's your opinion?

**Comment by richardjgowers** [comment]
Cool, thanks for reviewing, this is pushed to dev now.  I've tagged everything as 0.9.3 for now.
